pkg/meta/utils.go

```diff
@@ -369,3 +369,50 @@ func GetSummary(r Meta, ctx Context, inode Ino, summary *Summary, recursive bool
 	}
 	return 0
 }
+
+func FastGetSummary(r Meta, ctx Context, inode Ino, summary *Summary, recursive bool) syscall.Errno {
+	var attr Attr
+	if st := r.GetAttr(ctx, inode, &attr); st != 0 {
+		return st
+	}
+	if attr.Typ == TypeDirectory {
+		return fastGetSummary(r, ctx, inode, summary, recursive)
+	} else {
+		summary.Files++
+		summary.Length += attr.Length
+		summary.Size += uint64(align4K(attr.Length))
+	}
+	return 0
+}
+
+func fastGetSummary(r Meta, ctx Context, inode Ino, summary *Summary, recursive bool) syscall.Errno {
+	space, _, err := r.GetDirStat(ctx, inode)
```

Comment:
This should work in read-only mode

Reply:
In #3282, the `GetDirStat` add a `trySync` parameter, it works in read-only mode if we disable `trySync`.